src/dstack/_internal/core/backends/runpod/compute.py

```diff
@@ -120,6 +149,27 @@ def update_provisioning_data(self, provisioning_data: JobProvisioningData) -> No
                 provisioning_data.hostname = port["ip"]
                 provisioning_data.ssh_port = port["publicPort"]
 
+    def _generate_container_registry_auth_id(
+        self, registry_auth: Optional[RegistryAuth]
+    ) -> Optional[str]:
+        if registry_auth is None:
+            return None
+
+        return self.api_client.add_container_registry_auth(
+            uuid.uuid4().hex, registry_auth.username, registry_auth.password
+        )
+
+    def _clean_stale_container_registry_auths(self) -> None:
+        container_registry_auths = self.api_client.get_container_registry_auths()
+
+        # Container_registry_auths sorted by creation time so try to delete the oldest first
+        # when we reach container_registry_auths that is still in use, we stop
+        for container_registry_auth in container_registry_auths:
+            try:
+                self.api_client.delete_container_registry_auth(container_registry_auth["id"])
+            except Exception:
+                break
```

Comment:
I don't see a good reason to stop at the first failed delete. Maybe go over all the entries ignoring those that are in use?

Reply:
It doesn't make sense because if we encounter an error, it likely means that many of the following records are also still being used by some pods. Therefore, attempting to delete these records would be unnecessary work.

Reply:
> ignoring those that are in use

We can't know if they are in use without trying to delete them because there is a time window between pod termination and the unlocking of containerRegistryAuth for deletion.